src/taker/api.rs

```diff
@@ -328,6 +387,10 @@ impl Taker {
                         self.offerbook.add_bad_maker(bad_maker);
                     }
                     self.recover_from_swap()?;
+                    match handle.kill() {
+                        Ok(_) => log::info!("Tor instance terminated successfully"),
+                        Err(_) => log::error!("Error occured while terminating tor instance"),
+                    }
```

Comment:
There has to be a better place to put this logic, but there isn't currently. So the only option is to this awkward repeat of killing at every return site. 

A more clearer design could be 

``` rust
// Independent tor spawning routine
fn spawn_tor();

// The main public swap API
fn do_coinswap() {
    let tor_handle = self.spawn_tor()?;
    self.perform_swap()?;
    tor_handle.kill()?;
}
```

here `perform_swap()` will be all the code that `send_coinswap()` contains. And tor handling happens around it, not inside it.

`do_coinswap()` then becomes the main public api.